rcon/hooks.py

```diff
@@ -76,6 +89,97 @@ def initialise_vote_map(rcon: Rcon, struct_log):
         logger.exception("Something went wrong in vote map init")
 
 
+@on_chat
+def trigger_words(rcon: Rcon, struct_log: StructuredLogLineType):
+    config = TriggerWordsUserConfig.load_from_db()
+    if not config.enabled:
+        logger.debug("Trigger words are disabled")
+        return
+
+    chat_message = struct_log["sub_content"]
+    if chat_message is None:
+        return
+
+    event_cache = get_recent_actions()
+    steam_id_64: str = struct_log["steam_id_64_1"]
+    if steam_id_64 is None:
+        return
+
+    player_1_cache = event_cache.get(steam_id_64, MostRecentEvents())
+
+    player_nemesis_cache = (
+        event_cache[player_1_cache.last_nemesis_steam_id_64]
+        if player_1_cache.last_nemesis_steam_id_64
+        else MostRecentEvents()
+    )
+    player_victim_cache = (
+        event_cache[player_1_cache.last_victim_steam_id_64]
+        if player_1_cache.last_victim_steam_id_64
+        else MostRecentEvents()
+    )
+    player_tk_nemesis_cache = (
+        event_cache[player_1_cache.last_tk_nemesis_steam_id_64]
+        if player_1_cache.last_tk_nemesis_steam_id_64
+        else MostRecentEvents()
+    )
+    player_tk_victim_cache = (
+        event_cache[player_1_cache.last_tk_victim_steam_id_64]
+        if player_1_cache.last_tk_victim_steam_id_64
+        else MostRecentEvents()
+    )
+
+    chat_words = set(re.split(CHAT_WORDS_RE, chat_message))
```

Comment:
The second thing I have trouble to understand, tbh 😅 
I _think_ the only reason of this code is to separate words based to be able to look for trigger words, right? What I don't get is: Why would we separate `!who-killed-me` to `["!who", "killed", "me"]`? Shouldn't it be one entry in the array as the whole group of words? Like only splitting by spaces? It would be as simple as:
```
chat_words = chat_message.split(' ')
```

After all: why would you need to split the chat words at all? Wouldn't a simple "contains" for the trigger.words be sufficient? 🤔